### ISSUE TITLE
Dependencies between Event and Elements

### ISSUE THREAD

**Issue by BartAdv** [idris-hackers/iQuery#11]
I've tried to add `target` function to get an element that caused an event. The `Event` module does not see the `Elements` (it's the other way around at the moment). I am wondering whether it wouldn't be better for the `Event` to depend on `Elements` (this would require to move onXXX functions there), but this would also require to remove abstract from `Element`, as its constructor would now be used in other module.

So maybe it's better to just merge those two? Any other ideas?

**Comment by raichoo** [comment]
I don't think that removing `abstract` from `Element` would be a good idea. If you need to merge the two modules to get your stuff going, give it a shot. I'll check out the PR and will most likely merge it :) Go for it!

**Comment by BartAdv** [comment]
Yes, it would only expose the underlying `Ptr`. I will most likely merge them, but after some experiments with dependent events.
